### PR TITLE
Added spin animation utility

### LINKED ISSUE #11: Build spinner pattern/utility
Spec: https://github.com/ubuntudesign/vanilla-dashboard-theme-design/tree/master/Spinner

### PR DESCRIPTION
**Pull request by bartaz**
## Done

Added spin animation utility.

## QA

- Checkout this branch
- Run server ./run
- Go to animation examples page: http://0.0.0.0:8102/vanilla-dashboard-theme/examples/utilities/animations/
- Watch it spin

Or watch it spin on demo server directly: http://vanilla-dashboard-theme-pr-27.run.demo.haus/vanilla-dashboard-theme/examples/utilities/animations/

Spinner in different sizes:
Light background: http://vanilla-dashboard-theme-pr-27.run.demo.haus/vanilla-dashboard-theme/examples/base/icons/icons-light/
Dark background: http://vanilla-dashboard-theme-pr-27.run.demo.haus/vanilla-dashboard-theme/examples/base/icons/icons-dark/

## Details

Fixes #11 

## Screenshots

spin

### CODE REVIEW

**Comment by yaili** [comment]
@bartaz which types of review are needed? can you add labels?

**Comment by WillMoggridge** [comment]
I am finding this spinner makes my eyes feel a bit uneasy. Seems like it is coming from a combination of my eyes focusing the negative space and also the speed it spins at.

Perhaps a slower spin as standard is preferable?

I can raise this elsewhere if preferred but I wanted to flag it here.

File: scss/_utilities_animations.scss [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/// Animations
+@mixin theme-animations {
+  .u-animation--spin {
+    animation: spin map-get($animation-duration, sleepy) infinite linear !important;
```
Comment (Lukewh) [inline scss/_utilities_animations.scss]: Removing `!important` on the demo still works. If it's required maybe add a comment. If it's possible to remove it, please do!

**Comment by bartaz** [comment]
@Lukewh I've noticed that most utilities styles use in Vanilla `!important` - I guess as a mean to make sure they override any other styles, so that's why it's here as well.

But I agree that it shouldn't be used if not necessary, so I'm fine with removing it. Unless it is some vanilla standard to use `!important` in utilities.

**Comment by Lukewh** [comment]
@bartaz interesting - yeah I can see the value when these are added, however with an animation I think it's probably unlikely that they will be overriding another animation (and I think it would be bad if we were mixing animations)... I'll defer to @barrymcgee @yaili and @anthonydillon as I'm not sure what the right thing to do here is.

**Comment by bartaz** [comment]
@WillMoggridge I suppose you are talking about real demo (http://vanilla-dashboard-theme-pr-27.run.demo.haus/vanilla-dashboard-theme/examples/utilities/animations/) not a gif screenshot in description with bad frame rate?

@yaili Is this something we would like to address here? Adjust the speed or icon of the spinner not to make viewers dizzy?

**Comment by barrymcgee** [comment]
Utilities should always use `!important` as we always want to override everything else.

**Pull request merged**